SubArraySum.swift

```diff
@@ -0,0 +1,25 @@
+
+// Time Complexity : (O)n
+// Space Complexity : (O)1
+// Did this code successfully run on Leetcode : No. Wrong answer 
```

Comment:
Make use of hashing DS
store in map -> number of occurences of a particular element
iter thru arr and if running sum-k is found in map, add to count the mapped value of running sum-k